fiftyone/plugins/utils.py

```diff
@@ -219,3 +222,103 @@ def _do_get_plugin_info(task):
             spec = gh_repo
 
         logger.debug("Failed to retrieve plugin info for %s: %s", spec, e)
+
+
+def get_target_view(ctx, target):
+    """
+    Get the target view for the current context. This is a helper function that
+    returns the view based on the `target` parameter. To be used in conjunction
+    with `list_target_views()`.
+
+    Args:
+        ctx: the current context
+        target: the target view
+
+    Returns:
+        the target view
+
+    """
+    if target == "SELECTED_SAMPLES":
+        return ctx.view.select(ctx.selected)
+
+    if target == "DATASET":
+        return ctx.dataset
+
+    return ctx.view
+
+
+def list_target_views(ctx, inputs):
+    """
+    List the available target views for the current context. This is a helper
+    function that sets the `target` parameter based on the current context. Use
+    it in the `inputs` method of a plugin to allow the user to select the target
+    view from selected samples, the current view, or the entire dataset. To be
+    used in conjunction with `get_target_view()`.
+
+    Args:
+        ctx: the current context
+        inputs: the inputs object
+
+    Returns:
+        None
+
+    """
+    has_view = ctx.view != ctx.dataset.view()
+    has_selected = bool(ctx.selected)
+    default_target = "DATASET"
+    if has_view or has_selected:
+        target_choices = types.RadioGroup()
+        target_choices.add_choice(
+            "DATASET",
+            label="Entire dataset",
+            description="Run on the entire dataset",
+        )
+
+        if has_view:
+            target_choices.add_choice(
+                "CURRENT_VIEW",
+                label="Current view",
+                description="Run on the current view",
+            )
+            default_target = "CURRENT_VIEW"
+
+        if has_selected:
+            target_choices.add_choice(
+                "SELECTED_SAMPLES",
+                label="Selected samples",
+                description="Run on the selected samples",
+            )
+            default_target = "SELECTED_SAMPLES"
+
+        inputs.enum(
```

Comment:
This creates a `foo.types.Property` - instead of a utility function we should create a type for this. That along side a utility method on the `types.Object` class is how we've made these sorts of utilities in the past. I'd like to keep that consistent.

Reply:
Ah yeah, I think `types.Object` is actually the right tool for the job here 💡

Reply:
In the simplest case it might look like this:


```py
inputs.target_view(
    "target_view",
    label="Target view",
    description="Choose what samples to process",
)

target_view = ctx.params["target_view"]
```

And if you wanted to customize labels/descriptions and which types of views are allowed, perhaps it would look like this:

```py
from fiftyone.operators import types

choices = types.RadioGroup()
choices.add_choice(
    types.ViewTypes.DATASET,
    label="Entire dataset",
    description="Generate thumbnails for the entire dataset",
)
choices.add_choice(
    types.ViewTypes.VIEW,
    label="Current view",
    description="Generate thumbnails for the current view",
)
choices.add_choice(
    types.ViewTypes.SELECTED_SAMPLES,
    label="Selected samples",
    description="Generate thumbnails for the selected samples",
)

inputs.target_view(
    "target_view",
    label="Target view",
    description="Choose what samples to process",
    choices=choices,
)

# Allow access to target enum so plugin dev can conditionally do stuff below based on the specific choice that the user made?
d = ctx.params["target_view"]
assert type(d["view"]) == fo.DatasetView
assert type(d["target"]) == types.ViewTypes.SELECTED_SAMPLES
```

Starts to get a bit verbose, eh?